test/cqlpy/test_secondary_index.py

```diff
@@ -201,6 +202,343 @@ def test_create_index_oversized_name(cql, test_keyspace):
+
+        # Invalid type: string instead of a number.
+        do_syntax_test("bloom_filter_fp_chance = 'not a number'")
+        # FIXME: This is accepted. Probably related to scylladb/scylladb#2431.
+        # do_syntax_test("crc_check_chance = 'not a number'")
+        do_syntax_test("default_time_to_live = 'not a number'")
+        do_syntax_test("gc_grace_seconds = 'not a number'")
+        do_syntax_test("max_index_interval = 'not a number'")
+        do_syntax_test("memtable_flush_period_in_ms = 'not a number'")
+        do_syntax_test("min_index_interval = 'not a number'")
+
+        # Invalid number type: double instead of int.
+        # FIXME: Currently, Scylla accepts these values, but they should result in a syntax exception.
+        # do_syntax_test("default_time_to_live = 1.23")
+        # do_syntax_test("gc_grace_seconds = 1.23")
+        # do_syntax_test("max_index_interval = 4023.23")
+        # do_syntax_test("memtable_flush_period_in_ms = 1.23")
+        # do_syntax_test("min_index_interval = 1.23")
+        # do_syntax_test("speculative_retry = 1.23")
+
+        # Invalid value: probability out of range [0, 1].
+        do_configuration_error_test("bloom_filter_fp_chance = 1.23",
+                                    r"bloom_filter_fp_chance must be larger than 6.71e-05 and " \
+                                    r"less than or equal to 1.0 \(got 1.23\)")
+        do_configuration_error_test("bloom_filter_fp_chance = -0.5",
+                                    r"bloom_filter_fp_chance must be larger than 6.71e-05 and " \
+                                    r"less than or equal to 1.0 \(got -0.5\)")
+
+# Verify that creating an index with mismatched interval boundaries fails.
+def test_create_index_with_invalid_intervals(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int, u int") as table:
+        index_name = unique_name()
+        with pytest.raises(ConfigurationException, match="max_index_interval must be greater than min_index_interval"):
+            cql.execute(f"CREATE INDEX {index_name} ON {table}(v) " \
+                         "WITH max_index_interval = 2010 AND min_index_interval = 4020")
+
+# Verify that trying to create an index with an invalid memtable_flush_period_in_ms fails.
+def test_create_index_with_invalid_memtable_flush_period(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int, u int") as table:
+        index_name = unique_name()
+        errmsg = "memtable_flush_period_in_ms must be 0 or greater than 60000"
+        with pytest.raises(ConfigurationException, match=errmsg):
+            cql.execute(f"CREATE INDEX {index_name} ON {table}(v) WITH memtable_flush_period_in_ms = -70000")
+        with pytest.raises(ConfigurationException, match=errmsg):
+            cql.execute(f"CREATE INDEX {index_name} ON {table}(v) WITH memtable_flush_period_in_ms = 30000")
+
+
```

Comment:
A thought: You have here more than 400 lines of tests just for the different WITH attributes on a CREATE INDEX. All of them are very similar, all of them scylla_only. Perhaps it makes sense to split them into a new separate test file?